django_components/component.py

```diff
@@ -1,24 +1,46 @@
+from __future__ import annotations
+
 import warnings
+from contextlib import contextmanager
 from functools import lru_cache
+from typing import (
+    TYPE_CHECKING,
+    ClassVar,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    TypeVar,
+)
 
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 from django.forms.widgets import MediaDefiningClass
-from django.template.base import Node, TokenType
+from django.template import TemplateSyntaxError
+from django.template.base import Node, NodeList, Template
 from django.template.loader import get_template
 from django.utils.safestring import mark_safe
 
+from django_components import app_settings
+
 # Allow "component.AlreadyRegistered" instead of having to import these everywhere
 from django_components.component_registry import (  # noqa
     AlreadyRegistered,
     ComponentRegistry,
     NotRegistered,
 )
 
-TEMPLATE_CACHE_SIZE = getattr(settings, "COMPONENTS", {}).get(
-    "TEMPLATE_CACHE_SIZE", 128
-)
-ACTIVE_SLOT_CONTEXT_KEY = "_DJANGO_COMPONENTS_ACTIVE_SLOTS"
+if TYPE_CHECKING:
+    from django_components.templatetags.component_tags import (
+        FillNode,
+        SlotNode,
+    )
+
+
+T = TypeVar("T")
```

Comment:
So, the reason for TYPE_CHECKING is that without it an IDE will complain that it can't resolve FillNode annotations in function signatures. The TypeVar as I've used it here just a useful way to remind myself that a method returns a value of the same type as its receiver.

Reply:
Check! Happy to get types introduced.